common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -87,6 +92,26 @@ public static CalendarInterval fromString(String s) {
     }
   }
 
+  /**
+   * Convert a string to CalendarInterval. Unlike fromString, this method is case-insensitive and
+   * will throw IllegalArgumentException when the input string is not a valid interval.
+   *
+   * @throws IllegalArgumentException if the string is not a valid internal.
+   */
+  public static CalendarInterval fromCaseInsensitiveString(String s) {
```

Comment:
I'm ok with it, but eventually we should just use the parser, instead of reimplementing the parsing stuff.

Reply:
+1 for @cloud-fan 's comment.

Reply:
Also, cc @hvanhovell .

Reply:
Agreed. But it will be a new feature. I created https://issues.apache.org/jira/browse/SPARK-27753 to future improvement.